src/query/plan/operator.cpp

```diff
@@ -3333,6 +3339,10 @@ class AggregateCursor : public Cursor {
       // Aggregations skip Null input values.
       if (input_value.IsNull()) continue;
       const auto &agg_op = agg_elem_it->op;
+      if (agg_elem_it->distinct) {
+        if (unique_values_it->contains(input_value)) break;
+        unique_values_it->emplace(input_value);
```

Comment:
I think this is a good approach and this should work for all aggregations.

Reply:
```suggestion
        auto insert_result = unique_values_it->insert(input_value);
        if (insert_result.second) {
          break;
        }
```
This way we don't do unnecessary lookups on the hot-path.

Reply:
The container is an `unordered_set` and the lookup is done in O(1). I don't think it would be much difference in this particular case